subsys/net/l2/wifi/wifi_shell.c

```diff
@@ -1074,6 +1101,30 @@ static int cmd_wifi_ps_timeout(const struct shell *sh, size_t argc, char *argv[]
 	return 0;
 }
 
+#ifdef CONFIG_WIFI_NM_WPA_SUPPLICANT_CRYPTO_ENTERPRISE
+static int cmd_wifi_set_enterprise_creds(const struct shell *sh, size_t argc, char *argv[])
+{
+	struct net_if *iface = net_if_get_first_wifi();
+	struct wifi_enterprise_creds_params params = {0};
+
+	context.sh = sh;
+
+	params.ca_cert = (uint8_t *)ca_cert_test;
+	params.ca_cert_len = ca_cert_test_len;
+	params.client_cert = (uint8_t *)client_cert_test;
+	params.client_cert_len = client_cert_test_len;
+	params.client_key = (uint8_t *)client_key_test;
+	params.client_key_len = client_key_test_len;
```

Comment:
Zephyr has a certificate storage support, see `tls_credential` API https://docs.zephyrproject.org/latest/doxygen/html/group__tls__credentials.html. So instead of directly passing certs to around, couldn't we utilize the cert API for this?

Reply:
We have the same discussion in three threads, let's use https://github.com/zephyrproject-rtos/zephyr/pull/74847#discussion_r1705361899 as the main one.

Reply:
The proposed API using `struct wifi_enterprise_creds_params` looks like a public API to me and not for testing, or did I misunderstood this?

Reply:
Hi @jukkar The proposed API using struct wifi_enterprise_creds_params, it's not only used for testing in wifi shell, also used by wifi driver to set the enterprise certs when connecting with enterprise mode AP